pythran/pythonic/numpy/dot.hpp

```diff
@@ -1,60 +1,189 @@
 #ifndef PYTHONIC_NUMPY_DOT_HPP
 #define PYTHONIC_NUMPY_DOT_HPP
 
-#include "pythonic/numpy/asarray.hpp"
+#include "pythonic/types/ndarray.hpp"
 #include "pythonic/numpy/sum.hpp"
+#include "pythonic/types/numpy_expr.hpp"
 #include "pythonic/types/traits.hpp"
 
+#include <nt2/linalg/details/blas/mm.hpp>
+#include <nt2/linalg/details/blas/mv.hpp>
+
+template <class T>
+    struct is_blas_type : pythonic::types::is_complex<T>
+    {};
+
+template<>
+    struct is_blas_type<float> : std::integral_constant<bool, true>
+    {};
+
+template<>
+    struct is_blas_type<double> : std::integral_constant<bool, true>
+    {};
```

Comment:
On 27/03/2015 22:56, serge-sans-paille wrote:
>
> In pythran/pythonic/numpy/dot.hpp 
> <https://github.com/serge-sans-paille/pythran/pull/417#discussion_r27333981>:
>
> >  #include "pythonic/types/traits.hpp"
> >
> > +#include <nt2/linalg/details/blas/mm.hpp>
> > +#include <nt2/linalg/details/blas/mv.hpp>
> > +
> > +template <class T>
> > +    struct is_blas_type : pythonic::types::is_complex<T>
> > +    {};
> > +
> > +template<>
> > +    struct is_blas_type<float> : std::integral_constant<bool, true>
> > +    {};
> > +
> > +template<>
> > +    struct is_blas_type<double> : std::integral_constant<bool, true>
> > +    {};
>
> http://en.cppreference.com/w/cpp/types/is_floating_point
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/serge-sans-paille/pythran/pull/417/files#r27333981>.
>
I am not sure I can use it as it accepts long double while blas do not 
support it.